app/helpers/sessions_helper.rb

```diff
@@ -3,15 +3,37 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
```

Comment:
signed để làm gì e ?

Reply:
Sử dụng phương thức signed để viết và đọc cookie ạ

Reply:
remember_token không dùng signed vẫn viết và đọc được mà e ?

Reply:
khi không sử dụng signed thì đặt id dưới dạng văn bản thuần túy, phương thức này hiển thị hình thức của cookie ứng dụng ->kẻ tấn công dễ dàng xâm phạm tài khoản người dùng hơn. Để tránh sự cố này, sử dụng singed (đã ký), mã hóa an toàn cookie trước khi đặt nó trên trình duyệt